compiler/modules/nback/build.bal

```diff
@@ -1281,7 +1281,7 @@ function buildEquality(llvm:Builder builder, Scaffold scaffold, bir:EqualityInsn
                 return buildStoreBoolean(builder, scaffold, builder.iCmp(op, lhsValue, rhsValue), result);
             }
             else if reprIsString(lhsRepr) && reprIsString(rhsRepr) {
-                return buildEqualStringString(builder, scaffold, op, <llvm:PointerValue>lhsValue, <llvm:PointerValue>rhsValue, result);
+                return generateStringEquality(builder, scaffold, op, insn, lhsValue, rhsValue, result);
```

Comment:
Instead of `generateStringEquality`, separate out a `function isAnyOperandSmallString(bir:Operand[]) returns boolean`.

Reply:
Changed